src/v/cluster/scheduling/allocation_state.cc

```diff
@@ -163,4 +164,22 @@ result<uint32_t> allocation_state::allocate(model::node_id id) {
     return errc::node_does_not_exists;
 }
 
+allocation_state::node_rack_placement_t
+allocation_state::get_rack_placement() const {
```

Comment:
Can we do this logic in `score` method of constraint ? We have an access to the allocation node there. We do not have to do the lookup.